code/socialDistribution/views.py

```diff
@@ -203,17 +204,30 @@ def posts(request, author_id):
                 Category.objects.create(category=category, post=post)
 
 
+
+
+
         except ValidationError:
             context = get_home_context(author, True, "Something went wrong! Couldn't create post.")
             return render(request, 'home/index.html', context)
-        
+
         else:
             # if using view name, app_name: must prefix the view name
             # In this case, app_name is socialDistribution
             return redirect('socialDistribution:home', author_id=author_id)
     
     return render(request, 'posts/index.html')
 
+# https://www.youtube.com/watch?v=VoWw1Y5qqt8 - Abhishek Verma
+def likePost(request):
+    post = get_object_or_404(Post, id = request.POST.get('postId'))
+    author = Author.objects.get(user=request.user)
+    if post.likes.filter(id=author.id).exists():
+        post.likes.remove(author)
+    else:
+        post.likes.add(author)
```

Comment:
Let's move this functionality to the API path once it's done (I'm working on the API base rn and should have a PR up by tomorrow at the latest)